### DISCUSSION TITLE
HLD of mobile (MVP)

### DISCUSSION THREAD

**Discussion by ankitbko** [crimson-dev/crimson#4] [General]
Share HLD for the mobile app and its integration with storage account

**Comment by darkfusion90** [comment]
![Mobile-HLD](https://user-images.githubusercontent.com/35585319/120812408-93b12f00-c56a-11eb-89fb-11e03296fe84.png)
General design for Mobile

**Reply by darkfusion90** [reply]
![hld-step-2](https://user-images.githubusercontent.com/35585319/121191960-16e3c500-c88a-11eb-828c-c31b30d75620.png)
Slightly improved and divided into sub-components

**Comment by tarunsingh495** [comment]
![hld](https://user-images.githubusercontent.com/77376180/120812862-03271e80-c56b-11eb-895d-b5322fc2f902.png)

**Comment by ME-ON1** [comment]
LLD 

![lld3](https://user-images.githubusercontent.com/47457466/120831322-0b885500-c57d-11eb-8241-ccaa8714b92d.jpg)

**Reply by ankitbko** [reply]
Why does `Image` inherits from both `Status` and `Metadata`?

**Reply by ME-ON1** [reply]
I thought of adding `SyncStatus` information to the particular image, to know if that image is backed up or not. IMO this can be useful when we want indicate to the user about that pic, like google photos does ( fully backed pics cards are in solid color and currently backing up pics are blurred/diminished cards or failed has different color ) 
 `MetaData` every image has some meta data so can it can be used to get the information of the image.

**Reply by ankitbko** [reply]
Yes thats good idea. Do you think the relationship between these and `Image` should be of inheritance or composition?

**Reply by ME-ON1** [reply]
Ohh!! yes it is composition. As it wouldn't have a life if image doesn't exist ! will correct it !

**Comment by ankitbko** [comment]
![image](https://user-images.githubusercontent.com/3169316/121707481-3bdb7080-caf4-11eb-854b-3a478912f56f.png)
